smolyay/samples.py

```diff
@@ -508,3 +511,583 @@ def _create(self):
+        domain of the random points
+
+    num_points: int
+        number of points to generate
+
+    seed: int
+        seed for generating the random points
+
+    scramble : bool
+        Default True. Applies centering to Latin Hypercube points, Owen
+        scrambling to Halton points, and LMS+shift scrambling to Sobol points.
+
+    optimization : {None, "random-cd", "lloyd"}
+        Default None. If "random-cd" the coordinates of points are adjusted to
+        lower the centered discrepancy. If "lloyd", adjust points using a
+        Lloyd-Max algorithm to encourage even spacing.
+
+    bits : int
+        Default 30. Sets the max number of points that can be generated,
+        which is 2**bits.
+    """
+
+    def _create(self):
+        """Generate a set of random points
+
+        Generates a set of quasi-monte carlo Halton Sequence points.
+        """
+        lower_bounds = [bound[0] for bound in self.domain]
+        upper_bounds = [bound[1] for bound in self.domain]
+        p_gen = scipy.stats.qmc.Halton(
+            self.num_dimensions,
+            scramble=self.scramble,
+            optimization=self.optimization,
+            seed=self.seed,
+        ).random(n=self.num_points)
+        self._points = scipy.stats.qmc.scale(p_gen, lower_bounds, upper_bounds)
+
+
+class SobolRandomPointSet(_QMCRandomPointSet):
+    """Generates a grid using Sobol Sequence
+
+    Parameters
+    ----------
+    domain: list
+        domain of the random points
+
+    num_points: int
+        number of points to generate. Must be a power of 2.
+
+    seed: int
+        seed for generating the random points
+
+    scramble : bool
+        Default True. Applies centering to Latin Hypercube points, Owen
+        scrambling to Halton points, and LMS+shift scrambling to Sobol points.
+
+    optimization : {None, "random-cd", "lloyd"}
+        Default None. If "random-cd" the coordinates of points are adjusted to
+        lower the centered discrepancy. If "lloyd", adjust points using a
+        Lloyd-Max algorithm to encourage even spacing.
+
+    Raises
+    ------
+    ValueError
+        number of points must be a power of two
+    ValueError
+        number of points must be less than 2**bits
+    """
+
+    def __init__(
+        self, domain, num_points, seed, scramble=True, optimization=None, bits=30
+    ):
+        self._bits = 64  # max value of bits
+        super().__init__(domain, num_points, seed, scramble, optimization)
+        self.bits = bits
+
+    @property
+    def bits(self):
+        """{1, 2}: Whether to create an orthogonal array of points"""
+        return self._bits
+
+    @bits.setter
+    def bits(self, value):
+        bits = int(value)
+        if bits > 64:
+            raise ValueError("bits max value is 64")
+        elif bits < 1:
+            raise ValueError("bits must be at least 1")
+        if 2**bits < self.num_points:
+            raise ValueError("2**bits must be greater than number of points")
```

Comment:
It isn't required, but I like doing all error checking related to a value in `if-elif` so I remember they are connected.
```suggestion
        elif 2**bits < self.num_points:
            raise ValueError("2**bits must be greater than number of points")

```